base/src/main/java/proguard/optimize/inline/BaseLambdaInliner.java

```diff
@@ -0,0 +1,357 @@
+     * inliner.
+     *
+     * @return Returns a new method which has the lambda inlined in it, the lambda argument is also removed. If this
+     *         class was unable to inline the lambda into the method it will return null.
+     */
+    public Method inline() {
+        if (consumingMethod instanceof LibraryMethod)
+            return null;
+
+        ConstantInstruction c = lambda.constantInstruction();
+        lambda.clazz().constantPoolEntryAccept(c.constantIndex, new LambdaImplementationVisitor((programClass, programMethod, interfaceClass, bridgeDescriptor) -> {
+            this.interfaceClass = interfaceClass;
+            this.lambdaClass = programClass;
+            this.lambdaInvokeMethod = programMethod;
+            this.bridgeDescriptor = bridgeDescriptor;
+
+            if (!shouldInline(consumingClass, consumingMethod, programClass, lambdaInvokeMethod)) {
+                return;
+            }
+
+            // First we copy the method from the anonymous lambda class into the class where it is used.
+            MethodCopier copier = new MethodCopier(programClass, programMethod, BaseLambdaInliner.this);
+            consumingClass.accept(copier);
+        }));
+
+        return inlinedLambdaMethod;
+    }
+
+    /**
+     * A method that allows to conditionally inline lambdas based on information about the consuming method and the
+     * lambda implementation method.
+     * @param consumingClass The consuming class, this is the class that contains the consuming method.
+     * @param consumingMethod The consuming method (method taking the lambda as an argument).
+     * @param lambdaClass The class containing the implementation method of the lambda.
+     * @param lambdaImplMethod The implementation method of the lambda.
+     * @return Returns true if it should attempt to inline this lambda, if it's false it will stop and this lambda will
+     *         not be inlined.
+     */
+    protected abstract boolean shouldInline(Clazz consumingClass, Method consumingMethod, Clazz lambdaClass, Method lambdaImplMethod);
+
+    @Override
+    public void visitAnyMember(Clazz clazz, Member member) {
+        ProgramMethod method = (ProgramMethod) member;
+        method.u2accessFlags = (method.getAccessFlags() & ~AccessConstants.BRIDGE & ~AccessConstants.SYNTHETIC & ~AccessConstants.PRIVATE)
+                | AccessConstants.STATIC;
+
+        // Copy the invoke method
+        String invokeMethodDescriptor = method.getDescriptor(consumingClass);
+        DescriptorModifier descriptorModifier = new DescriptorModifier(consumingClass);
+        staticInvokeMethod = descriptorModifier.modify(method,
+            originalDescriptor -> {
+                // The method becomes static
+                String modifiedDescriptor = originalDescriptor.replace("(", "(Ljava/lang/Object;");
+                // Change return type if it has an effect on the stack size
```

Comment:
Can you add to the comment why this is necessary? Are you removing the boxing/unboxing as part of the inlining process?